dotcom-rendering/src/web/components/DynamicSlowMPU.tsx

```diff
@@ -24,8 +25,14 @@ type Props = {
 	containerPalette?: DCRContainerPalette;
 	showAge?: boolean;
 	index: number;
+	renderAds: boolean;
+	trails: DCRFrontCard[];
 };
 
+type MPUProps = Omit<Props, `renderAds` | `trails`>;
```

Comment:
nb. although we aren't using these sorts of type functions very widely in the codebase, they are present here and there. Even though it adds a bit of complexity here we thought that it was worthwhile because the props required by the MPU and non-MPU versions do diverge quite a bit, and having a single main `Props` type that we can use to define the other two prop types feels like it's safer than trying to maintain three separate types.

Reply:
The types we land up with are:
```ts
type MPUProps = {
	groupedTrails: DCRGroupedTrails;
	containerPalette?: DCRContainerPalette;
	showAge?: boolean;
	index: number;
};

type NonMPUProps = {
	containerPalette?: DCRContainerPalette;
	showAge?: boolean;
	trails: DCRFrontCard[];
};
```

These actually look quite different, and using the `Omit` infers that they might somehow be coupled.

I wonder if it might be best to just have separate Props for each container.

I'd probably locate the props closer to the Components as well making it easier to edit props in the file.

Reply:
to elaborate a bit on the above, I think the case for using `Omit` here is because there are certain props that are added to the parent component as a whole which are only being added in order to be funnelled to one layout or the other (i.e. `groupedTrails` and `index` in the first case, and `trails` in the second). I find that using `Omit` makes this clearer, but I can definitely see an argument the other way. Very happy to refactor as an explicit list of props for each if @Georges-GNM is 🙂

Reply:
Yup, that makes sense; I've made the adjustment.